File: app/src/main/java/com/leadinsource/dailyweightlog/app/DWLApplication.java
package com.leadinsource.dailyweightlog.app;

import android.app.Application;

import com.leadinsource.dailyweightlog.di.AppComponent;
import com.leadinsource.dailyweightlog.di.AppModule;
import com.leadinsource.dailyweightlog.di.DaggerAppComponent;

public class DWLApplication extends Application /* implements HasActivityInjector*/ {

    /*@Inject
    DispatchingAndroidInjector<Activity> dispatchingAndroidInjector;*/

    private static DWLApplication app;
    private AppComponent appComponent;

    @Override
    public void onCreate() {
        super.onCreate();

        app = this;

        appComponent = DaggerAppComponent
                .builder()
                .appModule(new AppModule(getApplicationContext()))
                .build();
        /*DaggerAppComponent
                .builder()
                .application(this)
                .build()
                .inject(this);*/
    }


    /*@Override
    public AndroidInjector<Activity> activityInjector() {
        return null;
    }*/

    public static DWLApplication app() {
        return app;
    }

    public AppComponent appComponent() {
        return appComponent;
    }
}

File: app/src/main/java/com/leadinsource/dailyweightlog/app/DWLApplication.kt
package com.leadinsource.dailyweightlog.app

import android.app.Application
import androidx.room.Room
import com.leadinsource.dailyweightlog.db.WeightDatabase

import com.leadinsource.dailyweightlog.di.AppComponent
import com.leadinsource.dailyweightlog.di.AppModule
import com.leadinsource.dailyweightlog.di.DaggerAppComponent

class DWLApplication : Application() {
    private val appComponent: AppComponent by lazy {
        DaggerAppComponent
                .builder()
                .appModule(AppModule(applicationContext))
                .build()
    }

    override fun onCreate() {
        super.onCreate()

        app = this

        Room.databaseBuilder(
            applicationContext,
            WeightDatabase::class.java, "weights"
        )

    }

    fun appComponent(): AppComponent {
        return appComponent
    }

    companion object {

        private lateinit var app: DWLApplication

        fun app(): DWLApplication {
            return app
        }
    }
}
